test/e2e/user_workloads_test.go

```diff
@@ -63,17 +63,17 @@ func (s *userWorkloadsTestSuite) TestIdler() {
 		require.NoError(s.T(), err)
 	}
 
-	// Noise pods are still there
+	// make sure that "noise" pods are still there
 	_, err = s.memberAwait.WaitForPods(idlerNoise.Name, len(podsNoise), wait.PodRunning(), wait.WithPodLabel("idler", "idler"))
 	require.NoError(s.T(), err)
 
 	// Create another pod and make sure it's deleted.
 	// In the tests above the Idler reconcile was triggered after we changed the Idler resource (to set a short timeout).
 	// Now we want to verify that the idler reconcile is triggered without modifying the Idler resource.
-	pod := s.createStandalonePod(idler.Name) // create just one standalone pod. No need to create all possible pod controllers which may own pods.
+	pod := s.createStandalonePod(idler.Name, "idler-test-pod-2") // create just one standalone pod. No need to create all possible pod controllers which may own pods.
 	// We can't really make sure that the pod was created first before the idler deletes it.
 	// So, let's just wait for three seconds assuming it will be enough for the API server to create a pod before start waiting for the Pod to be deleted.
-	time.Sleep(3 * time.Second)
+	time.Sleep(6 * time.Second)
```

Comment:
The explanation is just in the comment above :) :
```
// We can't really make sure that the pod was created first before the idler deletes it.
// So, let's just wait for three seconds assuming it will be enough for the API server to create a pod before start waiting for the Pod to be deleted.
```

Basically we are racing with the Idler. We create a pod which automatically deleted by the Idler (which has very short idling timeout). We might miss the pod in a wait because it has been already deleted by the idler.

Reply:
So, I'm not sure that sleeping 6 seconds vs. 3 seconds makes any difference. This sleep is not for waiting for the idler to kick in. It's for sleeping for some time which should be enough for the pod to be created. This sleep should not be affected by the idler timeout. Then when we think the pod should already be create we can start waiting for the idler to kill the pod.

Reply:
> The explanation is just in the comment above :)

It's not really obvious from the comment, but thanks for adding additional explanation. 
But still, why cannot we wait until the resource is present? Why should we miss the pod in a wait - the idler starts counting the timeout after the pod is created. The observation of the pod creation is the same for both the idler controller and for e2e test logic - the resource is just there - we can read it (get it).

Reply:
I increased the sleep duration in the test because I had some issues with the `job` which was deleted by the idler but its own pod was not deleted (and thus remained in the namespace, as an orphan, without any `ownerReferences`).
However, I later found that the deletion of the job pods could be ensured with the `PropagationPolicy` option (https://github.com/codeready-toolchain/member-operator/pull/205/files#diff-0ddf448ea61959821185b66811f19adbd7a7cc10c772d44722bc0cffa81a425bR339)

Reply:
that being said, I will take a look at @MatousJobanek 's proposal of waiting until the resource exists.

Reply:
done in 7882852